plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/telemetry/otel/OtelBaseTest.kt

```diff
@@ -0,0 +1,325 @@
+import java.util.concurrent.TimeUnit
+
+@ExtendWith(ApplicationExtension::class)
+class OtelBaseTest {
+    private companion object {
+        @RegisterExtension
+        val otelExtension = OtelExtension()
+    }
+
+    @Test
+    fun `context propagates from parent to child - happy case`() {
+        spanBuilder("tracer", "parentSpan").use {
+            spanBuilder("anotherTracer", "childSpan").use {}
+        }
+
+        assertThat(otelExtension.completedSpans).hasSize(2).satisfiesKt { spans ->
+            val parent = spans.first()
+            val child = spans.last()
+
+            assertThat(parent.parentSpanContext.traceId).isEqualTo(TraceId.getInvalid())
+            assertThat(child.parentSpanContext.traceId).isEqualTo(parent.spanContext.traceId)
+            assertThat(child.getAttribute(PLUGIN_ATTRIBUTE_KEY)).isEqualTo(parent.getAttribute(PLUGIN_ATTRIBUTE_KEY))
+        }
+    }
+
+    @Test
+    fun `context propagates from parent to child - happy case coroutines`() = runTest {
+        spanBuilder("tracer", "parentSpan").useWithScope {
+            spanBuilder("anotherTracer", "childSpan").useWithScope {}
+        }
+
+        assertThat(otelExtension.completedSpans).hasSize(2).satisfiesKt { spans ->
+            val parent = spans.first()
+            val child = spans.last()
+
+            assertThat(parent.parentSpanContext.traceId).isEqualTo(TraceId.getInvalid())
+            assertThat(child.parentSpanContext.traceId).isEqualTo(parent.spanContext.traceId)
+            assertThat(child.getAttribute(PLUGIN_ATTRIBUTE_KEY)).isEqualTo(parent.getAttribute(PLUGIN_ATTRIBUTE_KEY))
+        }
+    }
+
+    @Test
+    fun `context propagates from parent to child - with context override`() {
+        spanBuilder("tracer", "parentSpan").setParent(Context.current().with(AWS_PRODUCT_CONTEXT_KEY, AWSProduct.AMAZON_Q_FOR_VS_CODE)).use {
+            spanBuilder("anotherTracer", "childSpan").use {}
+        }
+
+        assertThat(otelExtension.completedSpans).hasSize(2).satisfiesKt { spans ->
+            val parent = spans.first()
+            val child = spans.last()
+
+            assertThat(parent.parentSpanContext.traceId).isEqualTo(TraceId.getInvalid())
+            assertThat(child.parentSpanContext.traceId).isEqualTo(parent.spanContext.traceId)
+            assertThat(parent.getAttribute(PLUGIN_ATTRIBUTE_KEY)).isEqualTo("Amazon Q For VS Code")
+            assertThat(child.getAttribute(PLUGIN_ATTRIBUTE_KEY)).isEqualTo(parent.getAttribute(PLUGIN_ATTRIBUTE_KEY))
+        }
+    }
+
+    @Test
+    fun `context propagates from parent to child when child overrides context`() {
+        spanBuilder("tracer", "parentSpan").use {
+            // parent->child relationship is still maintained because Context.current() will return parent context
+            spanBuilder("anotherTracer", "childSpan").setParent(Context.current().with(AWS_PRODUCT_CONTEXT_KEY, AWSProduct.AMAZON_Q_FOR_VS_CODE)).use {}
```

Comment:
Should we change this to AMAZON_Q_FOR_JETBRAINS?

Reply:
intentional so we know for sure it's working and not just picking up a default field somewhere